docs/docs/docs/01-core/database/03-migrations.md

```diff
@@ -0,0 +1,30 @@
+---
+title: Migrations
+description: Conceptual guide to migrations in Strapi
+tags:
+  - database
+  - migration
+---
+
+Strapi manages schema and data migrations in multiple ways. As much as possible we try to automatically sync the DB schema with the application configuration. However this in not sufficient to manage data migrations or schema migrations that are not reconcilable.
```

Comment:
Not against it, but I am wondering why user migrations are run before internal migrations?
From our last conversations, I recall that no matter the order we execute them conflicts could appear, but is there a benefit from running them in a specific order.

Reply:
- A users should have run all their custom migrations prior to upgrading strapi. (no more user migration to run) 
- Once upgrading a user will "receive" the new internal migrations.
- In order to give them the posbility to migrate some data around before we run strapi migrations I chose this order so it gives more control to users if need be. at least then can adapt to internal migrations this way if necessary. 

Thinking about v5 it might be a way for them to make certain custom changes that they will need before we run our migrations.